qiskit_machine_learning/utils/loss_functions/loss_functions.py

```diff
@@ -196,3 +261,56 @@ def gradient(self, predict: np.ndarray, target: np.ndarray) -> np.ndarray:
         return target * (1.0 / (1.0 + np.exp(-predict)) - 1) + (1 - target) * (
             1.0 / (1.0 + np.exp(-predict))
         )
+
+
+class SVCLoss(KernelLoss):
+    r"""
+    .. math::
+
+        \text{This class provides a kernel loss function for classification tasks by
+        fitting an ``SVC`` model from sklearn. Given training samples, x_{i}, with binary
+        labels, y_{i}, and a kernel, K_{θ}, parameterized by values theta, the loss is defined as}
+
+        SVCLoss = \sum_{i} a_i - 0.5 \sum_{i,j} a_i a_j y_{i} y_{j} K_θ(x_i, x_j)
+
+        \text{where a_i are the optimal Lagrange multipliers found by solving the standard
+        SVM quadratic program. Note that the hyper-parameter C for the soft-margin penalty can
+        be specified through the keyword args.}
+
+    Minimizing this loss over the parameters theta of the kernel is equivalent to maximizing a
+    weighted kernel alignment, which in turn yields the smallest upper bound to the SVM
+    generalization error for a given set of parameters.
+
+    See https://arxiv.org/abs/2105.03406 for further details.
+    """
+
+    def __init__(self, **kwargs):
+        self.kwargs = kwargs
+
+    def evaluate(
+        self,
+        parameter_values: Sequence[float],
+        quantum_kernel: 'QuantumKernel',
+        data: np.ndarray,
+        labels: np.ndarray,
+    ) -> float:
+        # Bind training parameters
+        quantum_kernel.assign_user_parameters(parameter_values)
+
+        # Train a quantum support vector classifier
+        svc = SVC(kernel=quantum_kernel.evaluate, **self.kwargs)
+        svc.fit(data, labels)
+
+        # Get dual coefficients
+        dual_coefs = svc.dual_coef_[0]
+
+        # Get support vectors
+        support_vecs = svc.support_
+
+        # Get estimated kernel matrix
+        kmatrix = quantum_kernel.evaluate(np.array(data))[support_vecs, :][:, support_vecs]
```

Comment:
more efficient if you restrict the data and then evaluate the kernel than if you first evaluate the full kernel matrix and then drop what's not needed.

Reply:
we should add an option to tell the kernel to cache results. Right now the kernel matrix is evaluated twice per iteration.

Reply:
We have addressed this by passing the kernel matrix into the SVC as a 'precomputed' kernel, which should prevent it from being evaluated twice.

We have some ideas about caching, but they are more involved and should probably go in another PR. Resolving this; will follow up about caching.